main.cpp

```diff
@@ -129,22 +130,26 @@ int run_assembly()
 	
 	list_free (include_dirs, true, NULL);
 	include_dirs = NULL;
-
+    if(firstPassFatalErrors >0){
+        exit(firstPassFatalErrors); //or should this be just a constant?

```

Comment:
This should be a constant. A return code greater than 255 is not very portable, so something like 256 errors might be reported back to a calling process as code 0 due to truncation.